solana/solana.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) The Amphitheatre Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package solana
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+
+	"github.com/buildpacks/libcnb"
+	"github.com/paketo-buildpacks/libpak"
+	"github.com/paketo-buildpacks/libpak/bard"
+	"github.com/paketo-buildpacks/libpak/effect"
+)
+
+type Solana struct {
+	LayerContributor libpak.DependencyLayerContributor
+	Logger           bard.Logger
+	Executor         effect.Executor
+}
+
+func NewSolana(dependency libpak.BuildpackDependency, cache libpak.DependencyCache) Solana {
+	contributor := libpak.NewDependencyLayerContributor(dependency, cache, libcnb.LayerTypes{
+		Build:  true,
+		Cache:  true,
+		Launch: true,
+	})
+	return Solana{
+		LayerContributor: contributor,
+		Executor:         effect.NewExecutor(),
+	}
+}
+
+func (r Solana) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
+	r.LayerContributor.Logger = r.Logger
+	return r.LayerContributor.Contribute(layer, func(artifact *os.File) (libcnb.Layer, error) {
+		return layer, nil
```

Comment:
miss code? Refer to: https://github.com/amp-buildpacks/hardhat/blob/main/hardhat/hardhat.go